src/cloudsc_gpu/cloudsc_gpu_scc_cuf_k_caching_mod.F90

```diff
@@ -0,0 +1,2628 @@
+    REAL(KIND=JPRB) :: ZVQX(NCLV)    ! fall speeds of three categories
+    REAL(KIND=JPRB) :: ZEXPLICIT
+    REAL(KIND=JPRB) :: ZRATIO(NCLV), ZSINKSUM(NCLV)
+    
+    ! for sedimentation source/sink terms
+    REAL(KIND=JPRB) :: ZFALLSINK(NCLV)
+    REAL(KIND=JPRB) :: ZFALLSRCE(NCLV)
+    
+    ! for convection detrainment source and subsidence source/sink terms
+    REAL(KIND=JPRB) :: ZCONVSRCE(NCLV)
+    REAL(KIND=JPRB) :: ZCONVSINK(NCLV)
+    
+    ! for supersaturation source term from previous timestep
+    REAL(KIND=JPRB) :: ZPSUPSATSRCE(NCLV)
+    
+    ! Numerical fit to wet bulb temperature
+    REAL(KIND=JPRB), PARAMETER :: ZTW1 = 1329.31_JPRB
+    REAL(KIND=JPRB), PARAMETER :: ZTW2 = 0.0074615_JPRB
+    REAL(KIND=JPRB), PARAMETER :: ZTW3 = 0.85E5_JPRB
+    REAL(KIND=JPRB), PARAMETER :: ZTW4 = 40.637_JPRB
+    REAL(KIND=JPRB), PARAMETER :: ZTW5 = 275.0_JPRB
+    
+    REAL(KIND=JPRB) :: ZSUBSAT    ! Subsaturation for snow melting term
+    REAL(KIND=JPRB) :: ZTDMTW0    ! Diff between dry-bulb temperature and
+    ! temperature when wet-bulb = 0degC
+    
+    ! Variables for deposition term
+    REAL(KIND=JPRB) :: ZTCG    ! Temperature dependent function for ice PSD
+    REAL(KIND=JPRB) :: ZFACX1I, ZFACX1S    ! PSD correction factor
+    REAL(KIND=JPRB) :: ZAPLUSB, ZCORRFAC, ZCORRFAC2, ZPR02, ZTERM1, ZTERM2    ! for ice dep
+    REAL(KIND=JPRB) :: ZCLDTOPDIST    ! Distance from cloud top
+    REAL(KIND=JPRB) :: ZINFACTOR    ! No. of ice nuclei factor for deposition
+    
+    ! Autoconversion/accretion/riming/evaporation
+    INTEGER(KIND=JPIM) :: IWARMRAIN
+    INTEGER(KIND=JPIM) :: IEVAPRAIN
+    INTEGER(KIND=JPIM) :: IEVAPSNOW
+    INTEGER(KIND=JPIM) :: IDEPICE
+    REAL(KIND=JPRB) :: ZRAINACC
+    REAL(KIND=JPRB) :: ZRAINCLD
+    REAL(KIND=JPRB) :: ZSNOWRIME
+    REAL(KIND=JPRB) :: ZSNOWCLD
+    REAL(KIND=JPRB) :: ZESATLIQ
+    REAL(KIND=JPRB) :: ZFALLCORR
+    REAL(KIND=JPRB) :: ZLAMBDA
+    REAL(KIND=JPRB) :: ZEVAP_DENOM
+    REAL(KIND=JPRB) :: ZCORR2
+    REAL(KIND=JPRB) :: ZKA
+    REAL(KIND=JPRB) :: ZCONST
+    REAL(KIND=JPRB) :: ZTEMP
+    
+    ! Rain freezing
+    LOGICAL :: LLRAINLIQ    ! True if majority of raindrops are liquid (no ice core)
+    
+    !----------------------------
+    ! End: new microphysics
+    !----------------------------
+    
+    !----------------------
+    ! SCM budget statistics
+    !----------------------
+    REAL(KIND=JPRB) :: ZRAIN
+    
+    REAL(KIND=JPRB) :: ZHOOK_HANDLE
+    REAL(KIND=JPRB) :: ZTMPL, ZTMPI, ZTMPA
+    
+    REAL(KIND=JPRB) :: ZMM, ZRR
+    REAL(KIND=JPRB) :: ZRG
+    
+    REAL(KIND=JPRB) :: ZZSUM, ZZRATIO
+    REAL(KIND=JPRB) :: ZEPSILON
+    
+    REAL(KIND=JPRB) :: ZCOND1, ZQP
+    
+    REAL(KIND=JPRB) :: PSUM_SOLQA
+
+
+
+    INTEGER(KIND=JPIM), VALUE, INTENT(IN) :: NGPBLKS
+    INTEGER(KIND=JPIM) :: IBL
+
+    INTEGER(KIND=JPIM) :: JK_I, JK_IP1, JK_IM1
+
+
+    JL = THREADIDX%X
+    IBL = BLOCKIDX%Z
+    
+    IF (IBL <= NGPBLKS .and. JL <= KLON) THEN
```

Comment:
Could you please put a comment here to explain why we mask the kernel body like this?

Reply:
Seems like this is only beneficial for the hoist version, thus I'll remove it.